src/wp-includes/l10n/class-wp-translation-controller.php

```diff
@@ -0,0 +1,410 @@
+<?php
+/**
+ * I18N: WP_Translation_Controller class.
+ *
+ * @package WordPress
+ * @subpackage I18N
+ * @since 6.5.0
+ */
+
+/**
+ * Class WP_Translation_Controller.
+ *
+ * @since 6.5.0
+ */
+class WP_Translation_Controller {
+	/**
+	 * Current locale.
+	 *
+	 * @since 6.5.0
+	 * @var string
+	 */
+	protected $current_locale = 'en_US';
+
+	/**
+	 * Map of loaded translations per locale and text domain.
+	 *
+	 * [ Locale => [ Textdomain => [ .., .. ] ] ]
+	 *
+	 * @since 6.5.0
+	 * @var array<string, array<string, WP_Translation_File[]>>
+	 */
+	protected $loaded_translations = array();
+
+	/**
+	 * List of loaded translation files.
+	 *
+	 * [ Filename => [ Locale => [ Textdomain => WP_Translation_File ] ] ]
+	 *
+	 * @since 6.5.0
+	 * @var array<string,array<string, array<string, WP_Translation_File|false>>>
+	 */
+	protected $loaded_files = array();
+
+	/**
+	 * Returns the WP_Translation_Controller singleton.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @return WP_Translation_Controller
+	 */
+	public static function instance() {
+		static $instance;
+
+		if ( ! $instance ) {
+			$instance = new WP_Translation_Controller();
```

Comment:
Would this be better?

```suggestion
			$instance = new self();
```